kpi-tracker/src/main.rs

```diff
@@ -485,215 +707,56 @@ async fn get_block_details(
 async fn process_block(
     node: &mut Client,
     block_hash: BlockHash,
-    db: &mut DB,
-) -> anyhow::Result<()> {
-    let mut accounts: BTreeMap<CanonicalAccountAddress, AccountDetails> = BTreeMap::new();
-    let mut account_transactions: BTreeMap<TransactionHash, TransactionDetails> = BTreeMap::new();
-    let mut contract_modules: BTreeMap<ModuleRef, ContractModuleDetails> = BTreeMap::new();
-    let mut contract_instances: BTreeMap<ContractAddress, ContractInstanceDetails> =
-        BTreeMap::new();
-    let mut transaction_account_relations: BTreeSet<TransactionAccountRelation> = BTreeSet::new();
-    let mut transaction_contract_relations: BTreeSet<TransactionContractRelation> = BTreeSet::new();
-
+) -> anyhow::Result<NormalBlockData> {
     let block_details = get_block_details(node, block_hash).await?;
     let block_events = get_block_events(node, block_hash).await?;
 
+    let mut block_data = NormalBlockData {
+        block_hash,
+        block_details,
+        accounts: HashMap::new(),
+        transactions: HashMap::new(),
+        contract_modules: HashSet::new(),
+        contract_instances: HashMap::new(),
+    };
+
     block_events
         .try_for_each(|be| {
             match be {
                 BlockEvent::AccountCreation(address, details) => {
-                    accounts.insert(CanonicalAccountAddress::from(address), details);
+                    block_data.accounts.insert(address, details);
```

Comment:
Same here, I would strongly recommend using a Vec to preserve the order of accounts as they are on the chain.